### PR TITLE
feat: add-new-chart-to-dashboard

### PR DESCRIPTION
**Pull request by rephus**
### Pre requirements:
- [x] I have followed (at least) the [PR section of the contributing guide](https://github.com/lightdash/lightdash/blob/main/.github/CONTRIBUTING.md#sending-a-pull-request).

### Reference:
Closes: #1665

### Description:
New modal for adding charts to dashboards (add new + add to existing dashboard) 

### Preview:
Screenshot from 2022-04-11 09-50-20

Peek 2022-04-11 09-36

Edge case: adding with no dashboards (will always show the "add new" first) 

Peek 2022-04-11 09-49

### Scope of the changes (select all that apply):
- [x] Frontend  
- [ ] Backend  
- [ ] Common  
- [ ] E2E tests  
- [ ] Docs  
- [ ] CI/CD

### CODE REVIEW

File: packages/frontend/src/components/SavedDashboards/AddTilesToDashboardModal.tsx [hunk 1]
```diff
@@ -1,118 +1,198 @@
...
+                                    dashboards
+                                        ? dashboards.map((dashboard) => ({
+                                              value: dashboard.uuid,
+                                              label: dashboard.name,
+                                          }))
+                                        : []
+                                }
+                            />
+                            <div
+                                style={{ cursor: 'pointer' }}
```
Comment (nathaliabj) [inline packages/frontend/src/components/SavedDashboards/AddTilesToDashboardModal.tsx:103]: I would prefer if we move this into an styled component, i understand why it is easier when it is just a 1 liner style, however one of the disadvantages of having inline style is that it can affect your page’s size and download time. it can also start looking messier and it is harder to maintain, as you might miss it when needing to update, and this will always override any other style.

We should only use inline styles when we don't have access to an element's stylesheet and we are trying to overwrite something.

**Review by nathaliabj — CHANGES_REQUESTED** [review]
Looks really good!, just some small updates on the inline styles.

**Review by nathaliabj — APPROVED** [review]
LGTM!! 🐑 IT!
